test/testvalueflow.cpp

```diff
@@ -1732,6 +1732,64 @@ class TestValueFlow : public TestFixture {
                "}";
         ASSERT_EQUALS(false, testValueOfX(code, 3U, 123));
 
+        code = "void f(int x) {\n"
+               "    if (x > 123) {\n"
+               "        a = x;\n"
+               "    }\n"
+               "}";
+        ASSERT_EQUALS(true, testValueOfX(code, 3U, 124));
+
+        code = "void f(int x) {\n"
```

Comment:
I don't see why code is reassigned

Reply:
That is what is done for the other tests. I was just following the same pattern.

Reply:
as far as I see you reassign the same string literal. where did I reassign the same string literal? I guess that is a mistake by me.

Reply:
Never mind, I see the problem.